### PR TITLE
Product detail page

### PR DESCRIPTION
**Pull request by ShekharDhangar**
Added Error 404 page .
Do not go with number of lines. As all other code is reviewed in previous PR, dont know why its still showing in this PR even after merging.
[LIVE-LINK-404-PAGE](https://shed-store-react-git-productdetailpage-shekhardhangar.vercel.app/404)
Just look into these below files for review.
[ERROR-PAGE-CSS](https://github.com/ShekharDhangar/shed-store-react/pull/6/files#diff-1bfe5190bd6d5f1e7b6711ca43e49b32c53c41cd6f6d749b6b63f8d5fb4a1c26)
[ERROR-PAGE](https://github.com/ShekharDhangar/shed-store-react/pull/6/files#diff-b1505db3509563e34b92a057eae8da99d4541f137afe278cdb16c280844ba544)

### CODE REVIEW

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
   return (
     <>
       <section className="flex nav-section">
         {menuRequired && (
           <div onClick={showSidebar} className="menu-icon">
-          <span></span>
-          <span></span>
-          <span></span>
-        </div>
+            <span></span>
```
Comment (MayankKumar10) [inline src/components/Navbar/Navbar.jsx]: remove the spans if it's not needed

File: src/components/ProductCard/ProductCard.css [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+.icon-badge .icon {
+  height: 2.1rem;
+}
+
+.product-card-container{
+  max-width: 32rem;
+}
+
+@media only screen and (max-width: 50em) {
```
Comment (MayankKumar10) [inline src/components/ProductCard/ProductCard.css:40]: try to use px in media queries

**Review by MayankKumar10 — APPROVED** [review]
Look into these suggestions else everything looks good to me you can merge it.

File: src/pages/Error404Page/Error404Page.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { useEffect,useState } from "react";
+export function Error404Page() {
+    const navigate = useNavigate();
+    const [counter, setCounter] = useState(10);
+    useEffect(() => {
+        counter > 0 && setTimeout(() =>setCounter(count=>count-1), 1000);
+        if(counter===0){
+            navigate("/")
+        }
+      }, [counter]);
```
Comment (BalajiHarinath) [inline src/pages/Error404Page/Error404Page.jsx]: Suggestion: It is a good practice to clear the previous setTimeout before the component mounts again.

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggested change everything looks good.

File: src/pages/Error404Page/Error404Page.jsx (hunk 3 above)
Comment (ShekharDhangar) [reply src/pages/Error404Page/Error404Page.jsx]: thanks didn't knew that, I was thinking what should I return in useEffect, but thankyou  I got it

File: src/components/Navbar/Navbar.jsx (hunk 1 above)
Comment (ShekharDhangar) [reply src/components/Navbar/Navbar.jsx]: Actually its need its for hamburger menu

**Pull request merged**